lib/pluginmanager/install_strategy_factory.rb

```diff
@@ -0,0 +1,23 @@
+# encoding: utf-8
+require "pluginmanager/ui"
+require "pluginmanager/install_strategy/elastic_xpack"
+require "pluginmanager/install_strategy/uri_xpack"
+
+module LogStash module PluginManager
+  class InstallStrategyFactory
+    AVAILABLES_STRATEGIES = [
+      LogStash::PluginManager::InstallStrategy::UriXpack,
+      LogStash::PluginManager::InstallStrategy::ElasticXpack,
+    ]
+
+    def self.create(plugins_arg)
+      AVAILABLES_STRATEGIES.each do |strategy|
+        if klass = strategy.match?(plugins_arg.first)
+          return klass
```

Comment:
We cannot use `#find` in this context because find will return the first occurrence that matched the predicate, in this case `LogStash::PluginManager::InstallStrategy::UriXpack`, using each and breaking early seems OK to retrieve the installer.